src/Symfony/Component/Messenger/Transport/Schedule/ScheduleTransportFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+
+class ScheduleTransportFactory implements TransportFactoryInterface
+{
+    public function __construct(private readonly ContainerInterface $schedulesLocator)
+    {
+    }
+
+    public function createTransport(string $dsn, array $options, SerializerInterface $serializer): ScheduleTransport
+    {
+        $parsedUrl = [];
+        if ('schedule://' !== $dsn && false === $parsedUrl = parse_url($dsn)) {
+            throw new InvalidArgumentException(sprintf('The given Schedule DSN "%s" is invalid.', $dsn));
+        }
+
+        $scheduleName = $options['name'] ?? $parsedUrl['host'] ?? 'default';
```

Comment:
```suggestion
        $scheduleName = $parsedUrl['host'] ?? 'default';
```

I don't see why we would support both.

Reply:
All bridges do it :man_shrugging: 

Tipically, a DSN is got from env vars, and it can partially overwrite defaults from options. It's very useful when an application is distributed to many nodes or/and env vars are changed in runtime.

Reply:
But here, we are configuring a "static" DSN where the host is a static/random name. I don't think this will ever be configured via a DSN and it will never be overridden. Or am I missing something?

Reply:
I agree with you :+1: Let's simplify it.